### PR TITLE
 fix help-wanted#13 

### PR DESCRIPTION
**Pull request by ANH25**
This fixes ankitects/help-wanted#13

the culprit is the `_processText` function in editor.py, which wraps the whole pasted text in anchor tags after checking if the **beginning** of the text looks like a URL (using `isURL`) and that it's not a media link.

I fixed that by processing the text line by line, checking for links and HTML-escaping it.

### CODE REVIEW

File: qt/aqt/editor.py [hunk 1]
```diff
@@ -1091,34 +1091,35 @@ def _processText(self, mime: QMimeData) -> Optional[str]:
...
-
-        # if there's more than one consecutive space,
-        # use non-breaking spaces for the second one on
-        def repl(match):
-            return match.group(1).replace(" ", "&nbsp;") + " "
-
-        txt = re.sub(" ( +)", repl, txt)
-
-        return txt
+        processed = ''
```
Comment (evandrocoan) [inline qt/aqt/editor.py:1094]: Using an array of strings `processed = [] & "".join(processed)` is exponentially faster than string concatenation: https://waymoot.org/home/python_string/ (Efficient String Concatenation in Python)

Comment (ANH25) [reply qt/aqt/editor.py]: done. Thanks!

**Comment by evandrocoan** [comment]
@ANH, you need to run `make fix` to fix the formatting and commit the change: https://github.com/ankitects/anki/runs/905447871#step:25:526

**Comment by dae** [comment]
Thanks ANH! I'll merge this once 2.1.29 is released.

**Pull request merged**